indexinglanguage/src/main/java/com/yahoo/vespa/indexinglanguage/expressions/HashExpression.java

```diff
@@ -0,0 +1,91 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.indexinglanguage.expressions;
+
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hashing;
+import com.yahoo.document.DataType;
+import com.yahoo.document.DocumentType;
+import com.yahoo.document.Field;
+import com.yahoo.document.datatypes.IntegerFieldValue;
+import com.yahoo.document.datatypes.LongFieldValue;
+import com.yahoo.document.datatypes.StringFieldValue;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Hashes a string value to a long or int (by type inference on the target value).
+ *
+ * @author bratseth
+ */
+public class HashExpression extends Expression  {
+
+    private final HashFunction hasher = Hashing.sipHash24();
```

Comment:
Does the hash function matter ? If so I guess we should make it configurable.
We have made bad choices for hash function in the past.

Reply:
Yes it matters, but if we make it configurable we leave the choice to somebody who's less well able to make it. Here I think performance matters less than minimizing collisions given real text and then siphash seems like a good choice,. although I'm not sure it is the very best. 

We can also make it configurable later, with this as the default, if the need surfaces.